### PR TITLE
Expose Web Push headers for CORS requests

### PR DESCRIPTION
**Pull request by kitcambridge**
This lets our tests make cross-origin requests directly.

### CODE REVIEW

File: autopush/endpoint.py [hunk 1]
```diff
@@ -261,7 +263,15 @@ def prepare(self):
         """Common request preparation"""
         if self.ap_settings.cors:
             self.set_header("Access-Control-Allow-Origin", "*")
-            self.set_header("Access-Control-Allow-Methods", self.cors_methods)
+            if self.cors_methods:
```
Comment (jrconlin) [inline autopush/endpoint.py]: Would we ever have occasion to not have values for any of these? If I recall, passing an empty header is equivalent to the header not being present.

File: autopush/tests/test_endpoint.py [hunk 2]
```diff
@@ -435,37 +435,55 @@ def handle_finish(result):
...
         endpoint = self.endpoint
         endpoint.ap_settings.cors = True
         endpoint.prepare()
         endpoint.options(None)
         eq_(endpoint._headers[ch1], "*")
-        eq_(endpoint._headers[ch2], "PUT")
+        eq_(endpoint._headers[ch2], "POST,PUT")
+        eq_(endpoint._headers[ch3], "content-encoding,encryption,"
+            "encryption-key,content-type")
+        eq_(endpoint._headers[ch4], "location")
```
Comment (jrconlin) [inline autopush/tests/test_endpoint.py]: Should we check that passing an empty arg does not set a header (if we're going to the effort to check?)

File: autopush/endpoint.py (hunk 1 above)
Comment (kitcambridge) [inline autopush/endpoint.py]: Good call.

**Comment by jrconlin** [comment]
r+

**Pull request merged**